packages/react/src/ResourcePropertyInput/ResourcePropertyInput.tsx

```diff
@@ -41,15 +51,19 @@ export interface ResourcePropertyInputProps {
 }
 
 export function ResourcePropertyInput(props: ResourcePropertyInputProps): JSX.Element {
-  const { property, name, defaultValue, onChange } = props;
+  const { property, name, onChange } = props;
   const defaultPropertyType =
     props.defaultPropertyType && props.defaultPropertyType !== 'undefined'
       ? props.defaultPropertyType
       : property.type[0].code;
-
   const propertyTypes = property.type as ElementDefinitionType[];
 
-  if ((property.isArray || property.max > 1) && !props.arrayElement) {
+  const isArrayInput = (property.isArray || property.max > 1) && !props.arrayElement;
```

Comment:
At this point, is `property` the `InternalElementDefinition` from the original `StructureDefinition`?  Or is it from the profile?  I'm just trying to wrap my head around which checks we may / may not need.

I'm pretty sure that `property.isArray` implies `property.max > 1`, but I'm not 100% sure how profiles fit into that.

Reply:
Checking both `property.isArray` and `property.max > 1` pre-dated all the profile work.

`isArray` [is defined](https://github.com/medplum/medplum/blob/b770cf9c8e50ed829ccb98b91805dc9ca975abb1/packages/core/src/typeschema/types.ts#L483) as `ed.base?.max ?? ed.max > 1` which lines up with [the spec](https://hl7.org/fhir/R4/profiling.html#cardinality):

> When a profile is constraining another profile where there are more cardinality options (e.g. low is not just 0 or 1, and high is not just 1 or *), the same principles still apply: the constraining profile can only allow what the base profile allows.

> Note that though a profile can constrain an element from x..* to x..1, this doesn't make any difference to the representation in the JSON format - the element will still be represented in an array. As an example, take Patient.name which has a cardinality of 0..*. In an unprofiled Patient, this will be represented as:

```javascript
{
  "resourceType" : "Patient",
	"name" : [{
	   "text" : "Peter James"
	}]
}
```

> Even if a profile is created on the resource that narrows the cardinality to 1..1, applications will still process the resource without knowledge of the profile. For this reason the representation will still be the same.

So we could simplify the check to the following if we wanted to.

```
const isArrayInput = property.isArray && !props.arrayElement;
```

Reply:
And to answer your original question, `property` could be coming from either a profile or the base `StructureDefinition`.